File: Programmers/Level_1/숫자문자열과영단어.cpp
#include <bits/stdc++.h>
using namespace std;

void init() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
}

int solution(string s) {
    int answer = 0, loop = 0;

    while (loop < 10) {
        switch (loop) {
            case 0:
                while (s.find("zero") != string::npos) {
                    int idx = s.find("zero");
                    s.replace(idx, 4, "0");
                }
                break;
            case 1:
                while (s.find("one") != string::npos) {
                    int idx = s.find("one");
                    s.replace(idx, 3, "1");
                }
                break;
            case 2:
                while (s.find("two") != string::npos) {
                    int idx = s.find("two");
                    s.replace(idx, 3, "2");
                }
                break;
            case 3:
                while (s.find("three") != string::npos) {
                    int idx = s.find("three");
                    s.replace(idx, 5, "3");
                }
                break;
            case 4:
                while (s.find("four") != string::npos) {
                    int idx = s.find("four");
                    s.replace(idx, 4, "4");
                }
                break;
            case 5:
                while (s.find("five") != string::npos) {
                    int idx = s.find("five");
                    s.replace(idx, 4, "5");
                }
                break;
            case 6:
                while (s.find("six") != string::npos) {
                    int idx = s.find("six");
                    s.replace(idx, 3, "6");
                }
                break;
            case 7:
                while (s.find("seven") != string::npos) {
                    int idx = s.find("seven");
                    s.replace(idx, 5, "7");
                }
                break;
            case 8:
                while (s.find("eight") != string::npos) {
                    int idx = s.find("eight");
                    s.replace(idx, 5, "8");
                }
                break;
            case 9:
                while (s.find("nine") != string::npos) {
                    int idx = s.find("nine");
                    s.replace(idx, 4, "9");
                }
                break;
            
            default:
                break;
        }
        loop++;
    }
    answer = stoi(s);
    return answer;
}

File: Programmers/Level_2/Swift_Code/Swift_Code/숫자문자열과영단어.swift
//
//  숫자문자열과영단어.swift
//  Swift_Code
//
//  Created by InJe Choi on 2022/05/03.
//

import Foundation

func solution(_ s: String) -> Int {
    let stringArr = s.map { String($0) }
    var strNum = "", tmp = ""
    let numbers = ["zero": "0", "one": "1", "two": "2", "three": "3", "four": "4",
                   "five": "5", "six": "6", "seven": "7", "eight": "8", "nine": "9"]
    
    for s in stringArr {
        if Int(s) != nil {
            strNum += s
        } else {
            tmp += s
        }
        
        if let num = numbers[tmp] {
            strNum += num
            tmp = ""
        }
    }

    return Int(strNum)!
}

// print(solution("2three45sixseven"))
